src/irr/asset/CImageWriterPNG.cpp

```diff
@@ -102,11 +102,11 @@ bool CImageWriterPNG::writeAsset(io::IWriteFile* _file, const SAssetWriteParams&
 	{
 		const auto channelCount = asset::getFormatChannelCount(imageView->getCreationParameters().format);
 		if (channelCount == 1)
-			convertedImage = IImageAssetHandlerBase::getTopImageDataForCommonWriting<asset::EF_R8_SRGB>(imageView);
+			convertedImage = IImageAssetHandlerBase::createImageDataForCommonWriting<asset::EF_R8_SRGB>(imageView);
 		else if(channelCount == 2 || channelCount == 3)
-			convertedImage = IImageAssetHandlerBase::getTopImageDataForCommonWriting<asset::EF_R8G8B8_SRGB>(imageView);
+			convertedImage = IImageAssetHandlerBase::createImageDataForCommonWriting<asset::EF_R8G8B8_SRGB>(imageView);
 		else
-			convertedImage = IImageAssetHandlerBase::getTopImageDataForCommonWriting<asset::EF_R8G8B8A8_SRGB>(imageView);
+			convertedImage = IImageAssetHandlerBase::createImageDataForCommonWriting<asset::EF_R8G8B8A8_SRGB>(imageView);
 	}
 	
 	const auto& convertedImageParams = convertedImage->getCreationParameters();
```

Comment:
`trueExtent` and `blockInfo` make no sense, you have all the info in the `convertedRegion`

Reply:
fixed in upcoming commit